ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java

```diff
@@ -181,8 +187,14 @@ private static void createGenerator(UcteNode ucteNode, VoltageLevel voltageLevel
         if (ucteNode.getPowerPlantType() != null) {
             switch (ucteNode.getPowerPlantType()) {
                 case C:
+                    voltageLevel.getNetwork().setProperty(UCTE_POWERPLANT_TYPE_KEY + generatorId, ucteNode.getPowerPlantType().toString());
+                    break;
                 case G:
+                    voltageLevel.getNetwork().setProperty(UCTE_POWERPLANT_TYPE_KEY + generatorId, ucteNode.getPowerPlantType().toString());
+                    break;
                 case L:
+                    voltageLevel.getNetwork().setProperty(UCTE_POWERPLANT_TYPE_KEY + generatorId, ucteNode.getPowerPlantType().toString());
+                    break;
```

Comment:
The simplest way to implement that, is to add a property to the generator whatever the type of the power plant. Add the property after the creation of the generator:
```
generator.setProperty(UCTE_POWER_PLANT_TYPE, ucte.getPowerPlantType());
```